src/components/SearchPage.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState, useEffect } from 'react'
+import Card from './Card'
+import Pagination from './Pagination'
+import { isCollected } from 'utils'
+
+type SearchPageProps = {
+  collection: Collection
+  dispatch: React.Dispatch<Action>
+}
+
+const SearchPage = ({ collection, dispatch }: SearchPageProps) => {
+  const { cards } = collection
+
+  const [formData, setFormData] = useState({
+    pokemonName: '',
+  })
+  const [responseData, setResponseData] = useState<ResponseDataType>({
+    data: [],
+    page: 1,
+    pageSize: 0,
+    count: 1,
+    totalCount: 0,
+  })
+  const [query, setQuery] = useState('')
```

Comment:
```suggestion
  const [searchTerm, setSearchTerm] = useState('')
```

I'd more explicit with the name here.